iceoryx2-cli/lib/src/config_descriptions.rs

```diff
@@ -0,0 +1,389 @@
+                    key: "defaults.publish-subscribe.enable-safe-overflow",
+                    value_type: "`true`|`false`",
+                    default: config.defaults.publish_subscribe.enable_safe_overflow.to_string(),
+                    description: "Default overflow behavior.",
+                },
+                CliEntry {
+                    key: "defaults.publish-subscribe.unable-to-deliver-strategy",
+                    value_type: "`Block`|`DiscardSample`",
+                    default: format!("{:?}", config.defaults.publish_subscribe.unable_to_deliver_strategy),
+                    description: "Default strategy for non-overflowing setups when delivery fails.",
+                },
+                CliEntry {
+                    key: "defaults.publish-subscribe.subscriber-expired-connection-buffer",
+                    value_type: "int",
+                    default: config.defaults.publish_subscribe.subscriber_expired_connection_buffer.to_string(),
+                    description: "Expired connection buffer size of the subscriber. Connections to publishers are expired when the publisher disconnected from the service and the connection contains unconsumed samples.",
+                },
+            ],
+        },
+        CliSection {
+            name: "Defaults: Event Messaging Pattern",
+            entries: vec![
+                CliEntry {
+                    key: "defaults.event.max-listeners",
+                    value_type: "int",
+                    default: config.defaults.event.max_listeners.to_string(),
+                    description: "Maximum number of listeners.",
+                },
+                CliEntry {
+                    key: "defaults.event.max-notifiers",
+                    value_type: "int",
+                    default: config.defaults.event.max_notifiers.to_string(),
+                    description: "Maximum number of notifiers.",
+                },
+                CliEntry {
+                    key: "defaults.event.max-nodes",
+                    value_type: "int",
+                    default: config.defaults.event.max_nodes.to_string(),
+                    description: "Maximum number of nodes.",
+                },
+                CliEntry {
+                    key: "defaults.event.event-id-max-value",
+                    value_type: "int",
+                    default: config.defaults.event.event_id_max_value.to_string(),
+                    description: "Greatest value an [`EventId`] can have.",
+                },
+                CliEntry {
+                    key: "defaults.event.deadline",
+                    value_type: "Option<Duration>",
+                    default: config.defaults.event.deadline.map_or("None".to_string(), |e| format!("{e:?}")),
+                    description: "Maximum allowed time between two consecutive notifications. If not sent after this time, all listeners attached to a WaitSet will be notified.",
+                },
```

Comment:
@elfenpiff @RTK-jangid AFAIK, if this key is not set, then the default value will be `None`, essentially deactivating the feature. But the user is also not supposed to use this like `deadline = Some(Duration)` but just `deadline = Duration`. I think the user needs to be made aware of this but I'm not sure what the best option would be. Maybe having `value_type: "Duration"` and mentioning either in `default` and/or `description` that the `deadline` will not be used if the key is not present in the config file. The same would apply to the other three options below.

Reply:
@elBoberido I think the points you are mentioning are completely valid but should not be part of this PR. I would recommend creating a new issue. Add all insights from this PR there and close all comments related to this here.

I think we should get this PR merged with the scope it has and all side discussions should be happening somewhere else - otherwise either this PR explodes or everyone is confused since issues are raised that have nothing to do with `iox2 config explain`